cdap-proto/src/main/java/co/cask/cdap/proto/ProtoTrigger.java

```diff
@@ -192,6 +195,75 @@ public String toString() {
     }
   }
 
+  /**
+   * Represents a program status trigger for REST requests/responses
+   */
+  public static class ProgramStatusTrigger extends ProtoTrigger {
+
+    protected final ProgramStatus programStatus;
+    protected final String namespace;
```

Comment:
the reason of separating these fields is that app name and version are not available in app configurer, but are decided only from app deployment info. In order to specify a program in the same app in this method `triggerOnProgramStatus(ProgramType programType, String program, ProgramStatus programStatus)`, app name and version have to be left as null first, and handled them later in `Scheduler`